### PR TITLE
Fixed WPS441 to allow variable names to be reused in multiple loops

### PR DESCRIPTION
**Pull request by huangyunzen**
# I have made things!

With code like:

```
         for widget in self._widgets:
             await widget.update()
         ...
         for widget in self._widgets:
             await widget.set_something(state)
```
 
WPS441 is triggered due to reusing the widget variable name in the second loop.
 
WPS441 shouldn't apply, as we are defining a new variable, as opposed to reusing the previous value from the last loop.

## Checklist

- [x] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues
#1416

### CODE REVIEW

File: wemake_python_styleguide/visitors/ast/blocks.py [hunk 1]
```diff
@@ -255,7 +255,8 @@ def _add_to_scope(self, node: ast.AST, names: Set[str]) -> None:
     def _check_variable_usage(self, node: ast.Name) -> None:
         context = cast(ast.AST, get_context(node))
         blocks = self._block_variables[context][node.id]
-        if all(is_contained_by(node, block) for block in blocks):
+
+        if not blocks or any(is_contained_by(node, block) for block in blocks):
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/blocks.py:259]: Let's create a new variables out of this condition. I am a bit lost reading it. 
We need some extra meta context.

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks!

File: wemake_python_styleguide/visitors/ast/blocks.py (hunk 1 above)
Comment (huangyunzen) [reply wemake_python_styleguide/visitors/ast/blocks.py]: Added a new variable and some context to understand it.

File: tests/test_visitors/test_ast/test_blocks/test_control_variables.py [hunk 2]
```diff
@@ -43,6 +43,14 @@ def wrapper():
     print(wrapper)
 """
 
+correct_for_multi_loops = """
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_blocks/test_control_variables.py:46]: Please, let's add some tests: I guess we need to test that `for` can reuse the variable this way. But other types of blocks cannot. Let's say that two `for` loops are ok, but a `for` loop followed by `with` block with the same variable names is not ok.

```python
wrong_loop_variable_reuse1 = """
def wrapper():
    for my_var in range(3):
        ...
    with some() as my_var:
        ...
"""
```

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks! I have looked through the tests and figured out that we don't have any tests that cover the case I have described above.

Let's add them! 👍

File: tests/test_visitors/test_ast/test_blocks/test_control_variables.py (hunk 2 above)
Comment (huangyunzen) [reply tests/test_visitors/test_ast/test_blocks/test_control_variables.py:46]: I have added the test you included, and added a variable that checks whether the block variable is reused in other types of blocks.

**Review by sobolevn — APPROVED** [review]
Thanks a lot!

**Pull request merged**